tests/prepare_test_data.py

```diff
@@ -0,0 +1,173 @@
+        'clone-url': 'https://github.com/pingcap/tidb',
+        'commits': ['482dc3f06c438c320e1fc64ff02a5479d2a989fb'],
+        'skip-dirs': ['Godeps'],
+    },
+    {
+        'name': 'github.com/cznic/ql',
+        'clone-url': 'https://github.com/cznic/ql',
+        'commits': ['77545ff365ff426dfab06c46d70fa77600cd55af'],
+        'skip-dirs': [],
+    },
+    {
+        'name': 'github.com/google/cayley',
+        'clone-url': 'https://github.com/google/cayley',
+        'commits': ['792fe5d42c4dd4d967c3db2e30bd380fcdfb3151'],
+        'skip-dirs': ['Godeps'],
+    },
+    {
+        'name': 'github.com/influxdata/influxdb',
+        'clone-url': 'https://github.com/influxdata/influxdb',
+        'commits': [
+            'f326c6c0e4349a6f9bca056571b01ded81a2887c',
+            '688bc7a2f1877b524a3c915ae1ff7981267fcb49',
+            'd771eb72f708c523bb2bec35a97f6bb83ae4ef1d',
+        ],
+        'skip-dirs': [],
+    },
+    {
+        'name': 'github.com/google/martian',
+        'clone-url': 'https://github.com/google/martian',
+        'commits': [
+            '4d4cb10a32e7f1237228d58ecf95c571b0947a11',
+            '1afb734cc2eb3a5dab708a901ee07106a94a9186',
+        ],
+        'skip-dirs': [],
+    },
+]
+
+
+def get_logger():
+    logger = logging.getLogger()
+    logger.setLevel(logging.INFO)
+    formatter = logging.Formatter('%(levelname)s: %(message)s')
+    handler = logging.StreamHandler()
+    handler.setFormatter(formatter)
+    logger.addHandler(handler)
+    return logger
+
+
+def prepare(destdir, projects):
+    logger = get_logger()
+    for project in projects:
+        logger.info('Preparing project "{}"'.format(project['name']))
+        basepath = os.path.join(destdir, project['name'])
+        try:
+            os.makedirs(basepath)
+        except OSError:
+            logger.error('Cannot create directory "{}"'.format(basepath))
+            continue
+        for commit in project['commits']:
+            logger.info('Processing commit "{}"'.format(commit))
+            path = os.path.join(basepath, commit)
+            data = {
+                'source_code_directory': path,
+                'directories_to_skip': project['skip-dirs'],
+                'project': project['name'],
+                'commit': commit,
+                'ipprefix': project['name'],
+            }
+            datafile = '{}.json'.format(path)
+            try:
+                with open(datafile, 'wb') as f:
+                    json.dump(data, f)
+            except IOError:
+                logger.error('Cannot open file "{}"'.format(datafile))
+                continue
+            try:
+                repo = git.Repo.clone_from(project['clone-url'], path)
+            except git.exc.GitCommandError:
+                logger.error('Cannot clone git repository "{}" to "{}"'.format(
+                    project['clone-url'], path))
+                continue
+            try:
+                archive = '{}.tar.gz'.format(path)
```

Comment:
What about to write all archives directly under destdir directory? And change archive's name to project-commit.tar.gz where you replace each occurrence of '\' and '.' with '-'? E.g.
```
github.com/cznic/ql
77545ff365ff426dfab06c46d70fa77600cd55af
```
to
```
github-com-cznic-ql-77545ff365ff426dfab06c46d70fa77600cd55af.tar.gz
```

At the end destdir should contain only tarballs and apis only.